src/partition/src/splitter.rs

```diff
@@ -277,7 +308,16 @@ mod tests {
         let insert = mock_insert_request();
         let rule = Arc::new(MockPartitionRule) as PartitionRuleRef;
         let spliter = WriteSplitter::with_partition_rule(rule);
-        let ret = spliter.split_insert(insert).unwrap();
+        let mock_schema = DataTypesSchema::new(vec![
+            ColumnSchema::new(
+                "enable_reboot",
+                ConcreteDataType::Boolean(BooleanType),
+                false,
+            ),
+            ColumnSchema::new("id", ConcreteDataType::Int16(Int16Type {}), false),
+            ColumnSchema::new("host", ConcreteDataType::String(StringType), true),
+        ]);
+        let ret = spliter.split_insert(insert, &mock_schema).unwrap();
```

Comment:
I realized your test; actually won't run into your code. In the test self.partition_rule.partition_columns() returns ["id"]. However, the `insert` id columns are non-null values.